src/main/java/practice/StreamPractice.java

```diff
@@ -75,6 +120,12 @@ public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
      * parametrized with Candidate in CandidateValidator.
      */
     public List<String> validateCandidates(List<Candidate> candidates) {
-        return Collections.emptyList();
+        Predicate<Candidate> candidatePredicate = new CandidateValidator();
+        List<String> candidatesNames = candidates.stream()
+                .filter(candidatePredicate::test)
```

Comment:
there is no need in calling `test` method, our validator implements `Predicate<Candidate>` so you can just pass it to `filter()`
```suggestion
                .filter(candidatePredicate)
```